kuma/wiki/forms.py

```diff
@@ -295,6 +305,34 @@ def clean_slug(self):
 
         return slug
 
+    def clean_tags(self):
+        """
+        Validate the tags ensuring we have no case-sensitive duplicates.
+        """
+        tags = self.cleaned_data['tags']
+        dupe_tags = []
+
+        if tags:
+            for tag in parse_tags(tags):
+                # Note: The exact match query doesn't work correctly with
+                # MySQL with regards to case-sensitivity. If we move to
+                # Postgresql in the future this code may need to change.
+                doc_tag = DocumentTag.objects.filter(
+                    name__exact=tag).values_list('name', flat=True)
+                # Write a log we can grep to help find duplicate document tags.
+                if len(doc_tag) > 1:
+                    log.warn('Found duplicate document tags: %s' % doc_tag)
+
+                if doc_tag:
```

Comment:
For this I like to use the [first](https://docs.djangoproject.com/en/1.8/ref/models/querysets/#first) queryset method nowadays since it sets the result to `None` if nothing is found automatically. So this could be written like this:

    doc_tag = (DocumentTag.objects.filter(name__exact=tag)
                                  .values_list('name', flat=True)
                                  .first())
    if doc_tag is not None:
       # ..

I hope this works with ValuesQuerySets as well at least..